test/browser/page.test.jsx

```diff
@@ -14,29 +14,57 @@ var FakeModal = React.createClass({
 });
 
 describe("page", function() {
-  var handler, page, xhr;
+  var handler, page, xhr, originalTitle;
 
-  beforeEach(function(done) {
-    xhr = sinon.useFakeXMLHttpRequest();
-    Router.run(routes.routes, '/', function(Handler) {
+  function visitPage(url, cb) {
+    Router.run(routes.routes, url, function(Handler) {
       handler = TestUtils.renderIntoDocument(<Handler/>);
       page = TestUtils.findAllInRenderedTree(handler, function(c) {
```

Comment:
Ah yeah, it's a bit confusing, as is my use of it here... Basically I was just trying to find the `<Page>` component in the heirarchy of rendered components that the router is giving us, but it is surprisingly hard to do that.

`findAllInRenderedTree` essentially craws through the entire component heirarchy, calling our test function, which returns `true` if the given component is one of the ones we're looking for. Since our `<Page>` component has a method called `showModal`, and it's the only component that has that method, we basically just test for that... It's not a great way of finding the `<Page>` component but I couldn't figure out anything else at the time. :/